code/modules/paperwork/ticketmachine.dm

```diff
@@ -0,0 +1,201 @@
+			qdel(ticket)
+		tickets.Cut()
+	update_icon()
+
+/obj/machinery/ticket_machine/Initialize(mapload)
+	. = ..()
+	update_icon()
+
+/obj/machinery/ticket_machine/proc/increment()
+	if(current_number > ticket_number)
+		return
+	if(current_number && !(emagged) && tickets[current_number])
+		tickets[current_number].audible_message("<span class='notice'>\the [tickets[current_number]] disperses!</span>")
+		qdel(tickets[current_number])
+	if(current_number < ticket_number)
+		current_number ++ //Increment the one we're serving.
+		playsound(src, 'sound/misc/announce_dig.ogg', 50, FALSE)
+		atom_say("Now serving ticket #[current_number]!")
+		if(!(emagged) && tickets[current_number])
+			tickets[current_number].audible_message("<span class='notice'>\the [tickets[current_number]] vibrates!</span>")
+		update_icon() //Update our icon here rather than when they take a ticket to show the current ticket number being served
+
+/obj/machinery/ticket_machine_button
+	name = "increment ticket counter"
+	desc = "Use this button after you've served someone to tell the next person to come forward."
+	icon = 'icons/obj/power.dmi'
+	icon_state = "crema_switch"
+	req_access = list()
+	var/id = 1
+	var/cooldown = FALSE
+
+/obj/machinery/ticket_machine_button/attack_ghost(mob/user)
+	if(user.can_advanced_admin_interact())
+		return attack_hand(user)
+
+/obj/machinery/ticket_machine_button/attack_hand(mob/user)
+	if(allowed(usr) || user.can_advanced_admin_interact())
+		for(var/obj/machinery/ticket_machine/M in world)
+			if(M.id == id)
+				if(cooldown)
+					return
+				cooldown = TRUE
+				M.increment()
+				addtimer(VARSET_CALLBACK(src, cooldown, FALSE), 10)
+	else
+		to_chat(usr, "<span class='warning'>Access denied.</span>")
+
+
+/obj/machinery/ticket_machine/update_icon()
+	switch(ticket_number) //Gives you an idea of how many tickets are left
+		if(0 to 49)
+			icon_state = "ticketmachine_100"
+		if(50 to 99)
+			icon_state = "ticketmachine_50"
+		if(100)
+			icon_state = "ticketmachine_0"
+	handle_maptext()
+
+/obj/machinery/ticket_machine/proc/handle_maptext()
+	switch(ticket_number) //This is here to handle maptext offsets so that the numbers align.
+		if(0 to 9)
+			maptext_x = 13
+		if(10 to 99)
+			maptext_x = 10
+		if(100)
+			maptext_x = 8
+	maptext = "[current_number]" //Finally, apply the maptext
+
+/obj/machinery/ticket_machine/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/hand_labeler_refill))
+		if(!(ticket_number >= max_number))
+			to_chat(user, "<span class='notice'>[src] refuses [I]! There [max_number-ticket_number==1 ? "is" : "are"] still [max_number-ticket_number] ticket\s left!</span>")
+			return
+		to_chat(user, "<span class='notice'>You start to refill [src]'s ticket holder (doing this will reset its ticket count!).</span>")
+		if(do_after(user, 30, target = src))
+			to_chat(user, "<span class='notice'>You insert [I] into [src] as it whirs nondescriptly.</span>")
+			qdel(I)
```

Comment:
put `user.drop_item()` right above the `qdel(I)`, just in case.

Reply:
Fixed